### DISCUSSION TITLE
Que .ino subir?

### DISCUSSION THREAD

**Discussion by Veroledez** [marcel-licence/esp32_basic_synth#56] [Q&A]
Hola, este repsoitorio y los demas son fantasticos, he visto cada uno  de tus videos y quiero probar esto, lamentablemente soy novato y no se que sketch o .ino subir, hay varios de estos en cada repositorio y no se cual subir al esp32 para que funcione.
Muchas gracias.

**Comment by marcel-licence** [comment]
Hi, thanks a lot. I've converted this into a discussion, beacuse issues are just for problems found in the code.
The .ino file having the name like the project is the one you are looking for. All other files shall be in the same directory and will be automatically opened by arduino.
Do you have hardware like shown in the videos?
Best regards

**Comment by Veroledez** [comment]
Hola, ya entiendo, los demas se abren y cargan automaticamente.
Por otra parte, se podra unir  dos proyectos, digamos el synth y el drum, o synth  looper, fusionarlos, o el esp32 no tiene tanto poder para lograrlo.
Muchas gracias

**Reply by marcel-licence** [reply]
Hi, you could combine projects but the implementation will require some more effort. In addition to that the processing power will be shared -> max polyphony will be reduced for example. The looper itself is not really power hungry (works like a delay). 
Best regards

**Reply by MichaelPNolan** [reply]
If you got one running well then sure, I intend to take more modules and add more hardware to see if I can for example stop the Basic Synth and use a sampler on the same software and use midi to switch module. Then perhaps have a couple of ESP32 with different midi channels running in parallel in a box that can be tasked to use combinations ...
However, if you are new to Arduino, i wouldn't get too far ahead until you find your way around one of the project ... make it work. I have to admit - this was one of the first projects that I had a bunch of files that open together because I'd been keeping it simple. If you save as to make versions it copies all the files in the directory and renames just one of the files for the new version.

My version, while I have an ESP32-A1S to try - I didn't have the midi to trigger it. 

First I built it up with a normal ESP32 + DAC + USBHost and blew up my Launchpad because of a power issue. Once I had the USBHost working and then I just finished building a midi-controller with another Arduino (pro micro) that has a combo of the 8 pots and ADC using a 4051 analog mux (8 channels) over in my midi-controller. Send that with notes. So I'm almost at the point of having all that Marcel has in some of the videos albeit not as complete until a storebought midi controller keyboard comes to give me more. I want to integrate a screen, LEDmatrix ... its a great project to learn from.
I have figured out the MidiMapping part now thanks to having my midi controller and that also helped me learn a bit more about controller messages and send the correct channel, controller numbers mapping to ADSR and FilterADSR - also tried his delay module that works well too. All works.
